# Information mapping on the text files

For dialogues_topic.txt: 

```
{
    1: Ordinary Life, 2: School Life, 3: Culture & Education,
    4: Attitude & Emotion, 5: Relationship, 6: Tourism , 7: Health, 
    8: Work, 9: Politics, 10: Finance
}
```
The number corresponds to the given **topic**.

---

For dialogues_act.txt:

```
{ 
    1: inform，2: question, 3: directive, 4: commissive 
}
```
The number corresponds to the **dialog act annotations** in dialogues_text.txt.

----

For dialogues_emotion.txt:

```
{ 
    0: no emotion, 1: anger, 2: disgust, 3: fear, 
    4: happiness, 5: sadness, 6: surprise
}
```

Each line corresponds to the **emotion annotations** in dialogues_text.txt.

In [2]:
def topic_mapping() -> dict:
    return {
        '1': 'Ordinary Life', '2': 'School Life', '3': 'Culture & Education',
        '4': 'Attitude & Emotion', '5': 'Relationship', '6': 'Tourism' , '7': 'Health', 
        '8': 'Work', '9': 'Politics', '10': 'Finance'
    }

def dialog_mapping() -> dict:
    return {
        '1': 'inform', '2': 'question', '3': 'directive', '4': 'commissive'
    }

def emotion_mapping() -> dict:
    return { 
        '0': 'no emotion', '1': 'anger', '2': 'disgust', '3': 'fear', 
        '4': 'happiness', '5': 'sadness', '6': 'surprise'
    }

In [35]:
"""
    Loading all the labels
"""

def read_label_file(label_file: str) -> dict:
    label_dict = {}
    
    with open(f"data\{label_file}") as file:
         for index, line in enumerate(file.readlines()):
            label_dict[index] = line.split()
            
    return label_dict

emotion_labels_dict = read_label_file('dialogues_emotion.txt')
print(len(emotion_labels_dict))

annotation_labels_dict = read_label_file('dialogues_act.txt')
print(len(annotation_labels_dict))

topic_labels_dict = read_label_file('dialogues_topic.txt')
print(len(topic_labels_dict))

13118
13118
13118


In [54]:
"""
    Loading the text file
"""

def read_utterances_text() -> dict:
    
    utterances_dict = {}
    
    with open("data\dialogues_text.txt", encoding='utf8') as file:
        for index,line in enumerate(file.readlines()):
            utterances = line.split('__eou__')
            
            # Remove the '\n' character
            utterances.pop()
            utterances_dict[index] = utterances
    
    return utterances_dict

utterances_dict = read_utterances_text()
print(len(utterances_dict))

13118


# Self Analysis

- The readme.txt says there is supposed to be 11,318 dialogues; however, the output from the text file is different. It shows 13,118